pkg/storage/edv/restprovider.go

```diff
@@ -0,0 +1,404 @@
+type MACDigester interface {
+	ComputeMAC(data []byte, kh interface{}) ([]byte, error)
+}
+
+// MACCrypto is used for computing MACs.
+type MACCrypto struct {
+	kh          interface{}
+	macDigester MACDigester
+}
+
+// ComputeMAC computes a MAC for data using a matching MAC primitive in kh.
+func (m *MACCrypto) ComputeMAC(data string) (string, error) {
+	dataMAC, err := m.macDigester.ComputeMAC([]byte(data), m.kh)
+	return string(dataMAC), err
+}
+
+// NewMACCrypto returns a new instance of a MACCrypto.
+func NewMACCrypto(kh interface{}, macDigester MACDigester) *MACCrypto {
+	return &MACCrypto{
+		kh:          kh,
+		macDigester: macDigester,
+	}
+}
+
+// RESTProvider is a store provider that can be used to store data in a server supporting the
+// data vault HTTPS API as defined in https://identity.foundation/secure-data-store/#data-vault-https-api.
+type RESTProvider struct {
+	vaultID                  string
+	macCrypto                *MACCrypto
+	indexKeyMACBase64Encoded string
+	restClient               restClient
+}
+
+// NewRESTProvider returns a new RESTProvider. edvServerURL is the base URL for the data vault HTTPS API.
+// vaultID is the ID of the vault where this provider will store data. The vault must be created in advance, and since
+// the EDV REST API does not provide a method to check if a vault with a given ID exists, any errors due to a
+// non-existent vault will be deferred until calls are actually made to it in the restStore.
+// macCrypto is used to create an encrypted indices, which allow for documents to be queries based on a key
+// without leaking that key to the EDV server.
+func NewRESTProvider(edvServerURL, vaultID string,
+	macCrypto *MACCrypto, httpClientOpts ...Option) (*RESTProvider, error) {
+	indexKeyMAC, err := macCrypto.ComputeMAC(indexName)
+	if err != nil {
+		return nil, fmt.Errorf(failComputeMACIndexName, err)
+	}
+
+	client := restClient{
+		edvServerURL: edvServerURL,
+		httpClient:   &http.Client{},
+	}
+
+	for _, opt := range httpClientOpts {
+		opt(&client)
+	}
+
+	return &RESTProvider{
+		vaultID:                  vaultID,
+		macCrypto:                macCrypto,
+		indexKeyMACBase64Encoded: base64.URLEncoding.EncodeToString([]byte(indexKeyMAC)),
+		restClient:               client,
+	}, nil
+}
+
+// OpenStore opens a new restStore, using name as the namespace.
+func (r *RESTProvider) OpenStore(name string) (storage.Store, error) {
+	return &restStore{
+		vaultID:                  r.vaultID,
+		name:                     name,
+		macCrypto:                r.macCrypto,
+		indexKeyMACBase64Encoded: r.indexKeyMACBase64Encoded,
+		restClient:               r.restClient,
+	}, nil
+}
+
+// CloseStore is not currently supported by RESTProvider.
+func (r *RESTProvider) CloseStore(string) error {
+	return errCloseStoreNotSupported
+}
+
+// Close is not currently supported by RESTProvider.
+func (r *RESTProvider) Close() error {
+	return errCloseStoreNotSupported
```

Comment:
This will return an error when calling [`framework.Aries.Close()`](https://github.com/hyperledger/aries-framework-go/blob/ca5e0650014e48021c33f0c96fe23a11123967cf/pkg/framework/aries/framework.go#L301-L324). Have it return `nil` instead. Same with `CloseStore()`.

Reply:
Fixed.